web/src/components/gnd-main/index.js

```diff
@@ -16,22 +16,47 @@
  */
 
 import React from 'react';
+import {Route} from 'react-router'; // react-router v4
 import GndMap from '../gnd-map';
 import GndHeader from '../gnd-header';
 import GndLegend from '../gnd-legend';
 import GndFeatureTypeEditor from '../gnd-feature-type-editor';
+import {connectProject, connectFeature} from '../../datastore.js';
+import {featurePath} from '../../route.js';
 
-class GndMain extends React.Component {
+function GndFeatureDetails({match}) {
+  return (
+    <div>
+      {/* TODO(maralka): Display feature data loaded in the store inside of
+        side panel. */}
+      Placeholder for {match.params.featureId} of project
+      {match.params.projectId}
+    </div>
+  );
+}
+
+GndFeatureDetails.propTypes = {
+  match: Object,
+};
+
+type Props = {
+  match: Object,
+};
+
+class GndMain extends React.Component<Props> {
   render() {
     return (
       <React.Fragment>
         <GndMap />
-        <GndHeader />
-        <GndLegend />
-        <GndFeatureTypeEditor />
+        <GndHeader projectId={this.props.match.params.projectId} />
+        <GndLegend projectId={this.props.match.params.projectId} />
+        <GndFeatureTypeEditor projectId={this.props.match.params.projectId} />
```

Comment:
Can we pass projectId in as a prop rather than digging into the match params directly?

In either case we should `const projectId = this.props.match.params.projectId`.